main.tf

```diff
@@ -15,50 +15,108 @@
  */
 
 resource "google_sql_database_instance" "default" {
-  name                 = "${var.name}"
-  project              = "${var.project}"
-  region               = "${var.region}"
-  database_version     = "${var.database_version}"
-  master_instance_name = "${var.master_instance_name}"
+  name                 = var.name
+  project              = var.project
+  region               = var.region
+  database_version     = var.database_version
+  master_instance_name = var.master_instance_name
 
   settings {
-    tier                        = "${var.tier}"
-    activation_policy           = "${var.activation_policy}"
-    authorized_gae_applications = ["${var.authorized_gae_applications}"]
-    disk_autoresize             = "${var.disk_autoresize}"
-    backup_configuration        = ["${var.backup_configuration}"]
-    ip_configuration            = ["${var.ip_configuration}"]
-    location_preference         = ["${var.location_preference}"]
-    maintenance_window          = ["${var.maintenance_window}"]
-    disk_size                   = "${var.disk_size}"
-    disk_type                   = "${var.disk_type}"
-    pricing_plan                = "${var.pricing_plan}"
-    replication_type            = "${var.replication_type}"
-    database_flags              = ["${var.database_flags}"]
-    availability_type           = "${var.availability_type}"
+    tier                        = var.tier
+    activation_policy           = var.activation_policy
+    authorized_gae_applications = var.authorized_gae_applications
+    disk_autoresize             = var.disk_autoresize
+    dynamic "backup_configuration" {
+      for_each = [var.backup_configuration]
+      content {
+        binary_log_enabled = backup_configuration["binary_log_enabled"]
+        enabled            = backup_configuration["enabled"]
+        start_time         = backup_configuration["start_time"]
```

Comment:
just a question, it should not be used through .value? Like backup_configuration.value["binary_log_enabled"]

Reply:
No, it should be used through .value, thank you!